python/cugraph/cugraph/experimental/datasets/dataset.py

```diff
@@ -118,22 +117,35 @@ def get_edgelist(self, fetch=False):
 
         return self._edgelist
 
-    def get_graph(self, fetch=False):
+    def get_graph(self, fetch=False, default_direction=True, weights=True):
         """
         Return a Graph object.
 
         Parameters
         ----------
         fetch : Boolean (default=False)
-            Automatically fetch for the dataset from the 'url' location within
-            the YAML file.
+            Downloads the dataset from the web.
+
+        default_direction : Boolean (defalut=True)
+            Follow the default 'directed' attribute of the dataset. If
+            False, it will return the undirected version of the graph.
+
+        weights : Boolean (default=True)
+            Include weights.
         """
         if self._edgelist is None:
             self.get_edgelist(fetch)
 
         self._graph = cugraph.Graph(directed=self.metadata['is_directed'])
-        self._graph.from_cudf_edgelist(self._edgelist, source='src',
-                                       destination='dst')
+        if (len(self.metadata['col_names']) > 2 and weights):
```

Comment:
If a user requests weights by passing `weights=True`, but the metadata doesn't have >2 columns, this will silently not add weights which might be confusing. 

Perhaps the arg should be `ignore_weights` which defaults to False. This is subtle but I think it helps make it more obvious that weighted graphs are returned whenever possible.

In either case, the docstring above should mention that weights will not be present if they aren't included in the dataset.

```suggestion
        if (len(self.metadata['col_names']) > 2 and not(ignore_weights)):
```